In [53]:
import pandas as pd
import numpy as numpy

from analyze_src.basic_data_inspection import DataInspection, DatatypesInspectionStrategy, SummaryInspectionStrategy
pd.set_option('display.max_columns',None)
pd.set_option('display.max_rows',500)

In [54]:
data_path = '../extracted_data/logistic_regression.csv'
df = pd.read_csv(data_path)

# Checking the basic code

In [55]:
# step1 basic Data inspection 

In [56]:
# initialize the data inspector with the strategy of datatype inspection
inspector = DataInspection(DatatypesInspectionStrategy())
inspector.execute_inspection(df)

Datatypes of the Nullcounts in the dataframe: 
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 396030 entries, 0 to 396029
Data columns (total 27 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   loan_amnt             396030 non-null  float64
 1   term                  396030 non-null  object 
 2   int_rate              396030 non-null  float64
 3   installment           396030 non-null  float64
 4   grade                 396030 non-null  object 
 5   sub_grade             396030 non-null  object 
 6   emp_title             373103 non-null  object 
 7   emp_length            377729 non-null  object 
 8   home_ownership        396030 non-null  object 
 9   annual_inc            396030 non-null  float64
 10  verification_status   396030 non-null  object 
 11  issue_d               396030 non-null  object 
 12  loan_status           396030 non-null  object 
 13  purpose               396030 non-null  object 
 14  title

In [49]:
#changing my strategy to summary inspection
inspector.set_strategy(SummaryInspectionStrategy())
inspector.execute_inspection(df)

Summary statistics of Numerical Columns:  
           loan_amnt       int_rate    installment    annual_inc  \
count  396030.000000  396030.000000  396030.000000  3.960300e+05   
mean    14113.888089      13.639400     431.849698  7.420318e+04   
std      8357.441341       4.472157     250.727790  6.163762e+04   
min       500.000000       5.320000      16.080000  0.000000e+00   
25%      8000.000000      10.490000     250.330000  4.500000e+04   
50%     12000.000000      13.330000     375.430000  6.400000e+04   
75%     20000.000000      16.490000     567.300000  9.000000e+04   
max     40000.000000      30.990000    1533.810000  8.706582e+06   

                 dti       open_acc        pub_rec     revol_bal  \
count  396030.000000  396030.000000  396030.000000  3.960300e+05   
mean       17.379514      11.311153       0.178191  1.584454e+04   
std        18.019092       5.137649       0.530671  2.059184e+04   
min         0.000000       0.000000       0.000000  0.000000e+00   
25% 

## Insights

1. Know your data
    
     - data contains 396030 datapoints with 26 columns

2. Summary statistics
    - There are 2 categorical values
    - Target variable is **Loan status** which have 318357 fully paid customers 

    **Loan amount**

    - Amount of loan applied by the borrower
    - Average of 14113.888089 loan is taken from LoanTap
    - Maximum of 40000, minimum of 500 is asked as loan by the borrower

    **term**

    - Term has 2 categories (36, 60) months
    - Leading 36 months with 302005 borrowers.
    
    **intrest rate**
    - Most of the loans are with interest rate of **13.639**
    - interest rates are in the range of 5.32 - 30.99 
